### ISSUE TITLE
PluginDependencyStrategyException - Considers 'latest' as an older version

### ISSUE THREAD

**Issue by karthikjayaraman80** [jenkinsci/plugin-installation-manager-tool#197]
Steps to recreate issue:
1. Create a plugins.txt file with the below 2 lines:
```
artifactory:3.7.3
gradle:latest
```

2. Run jenkins-plugin-cli. Verbose output below:
```
jenkins-plugin-cli -l --verbose --no-download -f plugins_test.txt
File containing list of plugins to be downloaded: plugins_test.txt
Reading in plugins from plugins_test.txt

No directory to download plugins entered. Will use default of /usr/share/jenkins/ref/plugins
Using update center https://updates.jenkins.io/update-center.json from JENKINS_UC environment variable
Using experimental update center https://updates.jenkins.io/experimental/update-center.json from JENKINS_UC_EXPERIMENTAL environment variable
Using incrementals mirror https://repo.jenkins-ci.org/incrementals from JENKINS_INCREMENTALS_REPO_MIRROR environment variable
No CLI option or environment variable set for plugin info, using default of https://updates.jenkins.io/plugin-versions.json
No war entered. Will use default of /usr/share/jenkins/jenkins.war
Jenkins version: 2.249.1

Retrieving update center information
Returning cached value for: update-center-2.249.1
Returning cached value for: experimental-update-center-2.249.1
Returning cached value for: plugin-versions

artifactory depends on:
maven-plugin 3.7
workflow-cps 2.83
workflow-job 2.40
workflow-step-api 2.22
ant 1.11
apache-httpcomponents-client-4-api 4.5.10-2.0
config-file-provider 3.7.0
credentials 2.3.13
gradle 1.36
ivy 2.1
junit 1.37
mailer 1.32.1
matrix-project 1.17
run-condition 1.3
ssh-credentials 1.18.1
token-macro 2.12
io.jenkins.tools.pluginmanager.impl.PluginDependencyStrategyException: Plugin artifactory:3.7.3 depends on gradle:1.36, but there is an older version defined on the top level - gradle:latest
        at io.jenkins.tools.pluginmanager.impl.PluginManager.resolveRecursiveDependencies(PluginManager.java:843)
        at io.jenkins.tools.pluginmanager.impl.PluginManager.findPluginsAndDependencies(PluginManager.java:484)
        at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:147)
        at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:112)
        at io.jenkins.tools.pluginmanager.cli.Main.main(Main.java:63)
Plugin artifactory:3.7.3 depends on gradle:1.36, but there is an older version defined on the top level - gradle:latest
```

The code incorrectly considers latest as an older version to 1.36. However, flipping the dependencies in the plugins text file works correctly.
```
gradle:latest
artifactory:3.7.3
```

Note that this is a simplistic file and we have a huge number of plugins that we want to automatically install in our jenkins instances and would like to list our plugins in alphabetical order for better readability and management.

**Comment by raspy** [comment]
This has already been fixed with #184 and not reproducible with version 2.1.0.

**Comment by torstenwalter** [comment]
@timja This one can be closed As #184 fixed it.
